projects/batfish-common-protocol/src/main/java/org/batfish/common/plugin/PluginConsumer.java

```diff
@@ -90,17 +101,31 @@ public PluginConsumer(boolean serializeToText, List<Path> pluginDirs) {
     }
   }
 
-  public <S extends Serializable> S deserializeObject(Path inputFile, Class<S> outputClass) {
-    byte[] data = fromGzipFile(inputFile);
-    return deserializeObject(data, outputClass);
+  protected <S extends Serializable> S deserializeObject(Path inputFile, Class<S> outputClass) {
+    try {
+      // Awkward nested try blocks required because we refuse to throw IOExceptions.
+      try (Closer closer = Closer.create()) {
+        FileInputStream fis = closer.register(new FileInputStream(inputFile.toFile()));
+        GZIPInputStream gis = closer.register(new GZIPInputStream(fis));
+        return deserializeObject(gis, outputClass);
+      }
+    } catch (IOException e) {
+      throw new BatfishException(
+          String.format(
+              "Failed to deserialize object of type %s from file %s",
+              outputClass.getCanonicalName(), inputFile),
+          e);
+    }
   }
 
   protected byte[] fromGzipFile(Path inputFile) {
     try {
-      FileInputStream fis = new FileInputStream(inputFile.toFile());
-      GZIPInputStream gis = new GZIPInputStream(fis);
-      byte[] data = IOUtils.toByteArray(gis);
```

Comment:
This file stream was leaked -- IOUtils.toByteArray does not close the input stream.